src/Microsoft.AspNetCore.Routing/Matching/JumpTableBuilder.cs

```diff
@@ -19,12 +21,18 @@ internal class JumpTableBuilder
         // case because parameters don't match a zero-length segment.
         public int ExitDestination { get; set; } = InvalidDestination;
 
-        public void AddEntry(string text, int destination)
+        public void SetPathEntries((string text, int destination)[] entries)
         {
-            _entries.Add((text, destination));
+            _pathEntries = entries;
         }
 
-        public JumpTable Build()
+        public void SetPolicyEntries(INodeBuilderPolicy nodeBuilder, PolicyJumpTableEdge[] entries)
+        {
+            _nodeBuilder = nodeBuilder;
+            _policyEntries = entries;
+        }
```

Comment:
Ok this has definitely got wierd. What's the value of this vs just making `BuildPath` a static method?

Reply:
Static where? I put these fields on JumpTableBuilder because we need both the policy edges plus the node builder, and it felt cleaner to have here and have an array of JumpTableBuilders than have a tuple array of  JTB, node builder and policy edges.

Reply:
Ah I see.

I super don't get why these have become a single class :(

Reply:
What about getting rid of JumpTableBuilder altogether? I suspect you mostly did it because you didn't know the exit point until all DfaNodes had been created. Now that the exit is known beforehand the JumpTable and PolicyJumpTable could be created as we go.